processing/data_collection/gazette/spiders/sp_guaruja.py

```diff
@@ -8,30 +10,88 @@
 
 
 class SpGuaruja(BaseGazetteSpider):
+
     TERRITORY_ID = "3518701"
     name = "sp_guaruja"
     allowed_domains = ["guaruja.sp.gov.br"]
-    start_urls = ["http://www.guaruja.sp.gov.br/index.php/diario-oficial/"]
+    start_urls = ["http://www.guaruja.sp.gov.br/edicoes-diario-oficial/"]
+
+    def __init__(self, start_date=None, end_date=None, *args, **kwargs):
+        self.start_date = dt.date(2009, 1, 1)
+        self.end_date = dt.date.today()
+
+        super(SpGuaruja, self).__init__(start_date, end_date, *args, **kwargs)
+
+        self.logger.debug(
+            "Start date is {date}".format(date=self.start_date.isoformat())
+        )
+        self.logger.debug("End date is {date}".format(date=self.end_date.isoformat()))
 
     def parse(self, response):
-        months = response.css("div.span12 a::attr(href)").extract()
-        for month_url in months:
-            yield scrapy.Request(month_url, self.parse_items)
-
-    def parse_items(self, response):
-        gazettes = response.css("div.span12 p")
-        for gazette in gazettes:
-            date = gazette.css("a ::text").extract_first()
-            is_extra_edition = "parte" in date and "parte1" not in date
-            extra_editions = [is_extra_edition, True]
-            date = parse(date, languages=["pt"]).date()
-            urls = gazette.css("a::attr(href)").extract()
-            for url, is_extra_edition in zip(urls, extra_editions):
-                yield Gazette(
-                    date=date,
-                    file_urls=[url],
-                    is_extra_edition=is_extra_edition,
-                    territory_id=self.TERRITORY_ID,
-                    power="executive_legislature",
-                    scraped_at=datetime.utcnow(),
-                )
+        calendar_id = response.xpath(
+            '//div[@class="mec-search-form mec-totalcal-box"]/div/@class'
+        ).get()
+
+        # This ID is prefixed by 'mec_search_form_'. IE: mec_search_form_18101
+        calendar_id = calendar_id.replace("mec_search_form_", "")
+
+        target_date = self.start_date
+        data = {
+            "action": "mec_full_calendar_switch_skin",
+            "skin": "monthly",
+            "atts[skin]": "full_calendar",
+            "atts[show_past_events]": "1",
+            "atts[id]": calendar_id,
+            "apply_sf_date": "1",
+        }
+
+        while target_date <= self.end_date:
+            data.update(
+                {
+                    "sf[month]": f"{target_date.month:02}",
+                    "sf[year]": str(target_date.year),
+                }
+            )
+            yield scrapy.FormRequest(
+                url="https://www.guaruja.sp.gov.br/wp-admin/admin-ajax.php",
+                formdata=data,
+                method="POST",
+                callback=self.parse_events,
+            )
+            target_date = target_date + relativedelta(months=1)
```

Comment:
This is not wrong. But just sharing another possible approach. Some [spiders](https://github.com/okfn-brasil/querido-diario/blob/6d6cb229e9a8ec4a91ff85f2a98dd4861a1557b5/processing/data_collection/gazette/spiders/sp_guarulhos.py#L19) also use `rrule` to iterate over dates.

Reply:
This is not a request change. I just selected the wrong option when submitted the review. :facepalm: